shell/platform/windows/window_win32.cc

```diff
@@ -373,8 +373,10 @@ WindowWin32::HandleMessage(UINT const message,
     }
     case WM_SETFOCUS:
       ::CreateCaret(window_handle_, nullptr, 1, 1);
+      OnSetFocus();
       break;
     case WM_KILLFOCUS:
+      OnKillFocus();
```

Comment:
Please add tests that check that `OnSetFocus` and `OnKillFocus` are called for their respective messages in `shell/platform/windows/window_win32_unittests.cc`

Reply:
done